backend/uclapi/dashboard/api_applications.py

```diff
@@ -183,3 +207,152 @@ def delete_app(request):
             "success": True,
             "message": "App sucessfully deleted.",
         })
+
+
+def set_callback_url(request):
+    if request.method != "POST":
+        return HttpResponseBadRequest(json.dumps({
+            "success": False,
+            "error": "Request is not of method POST"
+        }))
+    try:
+        app_id = request.POST["app_id"]
+        new_callback_url = request.POST["callback_url"]
+        user_id = request.session["user_id"]
+    except KeyError:
+        response = PrettyJsonResponse({
+            "success": False,
+            "message": "Request does not have app_id."
+        })
+        response.status_code = 400
+        return response
+
+    # Check if the new callback URL uses an acceptable protocol
+    # (e.g. HTTP or HTTPS)
+    # The list comprehension will explode the
+    # UCLAPI_OAUTH_CALLBACK_ALLOWED_PROTOCOLS environment
+    # variable, split by semicolons, append each one with :// then
+    # check if the URL starts with it.
+    # If none of them work then bail out.
+    if not any([new_callback_url.startswith(p + "://") for p in os.environ.get(
+            "UCLAPI_OAUTH_CALLBACK_ALLOWED_PROTOCOLS").split(';')]):
+        response = PrettyJsonResponse({
+            "success": False,
+            "message": ("The requested callback URL"
+                        " does not use an acceptable protocol.")
+        })
+        response.status_code = 400
+        return response
+
+    url_data = tldextract.extract(new_callback_url)
+
+    if any(
```

Comment:
Have you seen how I did url safety verification for webhooks?

https://github.com/uclapi/uclapi/blob/f926598a10b86d0633de83959f6dcfaf1ae2a21c/backend/uclapi/dashboard/webhook_views.py#L45-L57

It seems slightly easier and more concise than the method used here

Reply:
This looks quite nice, and I can improve that with the environment variables. If I extract that out into an api_application then when we rebase webhooks on OAuth you could use that function?

Reply:
yep, that sounds great 💖